docs/book/ceph-authorization.md

```diff
@@ -0,0 +1,1197 @@
+
+    if len(sys.argv) == 4:
+        bucket_name = sys.argv[3]
+
+    access_key_env_name = '{}_ACCESS_KEY'.format(user.upper())
+    secret_key_env_name = '{}_SECRET_KEY'.format(user.upper())
+
+    access_key = os.getenv(access_key_env_name)
+    secret_key = os.getenv(secret_key_env_name)
+
+    conn = boto.connect_s3(
+            aws_access_key_id=access_key,
+            aws_secret_access_key=secret_key,
+            host=os.getenv("HOST"), port=int(os.getenv("PORT")),
+            is_secure=False, calling_format=boto.s3.connection.OrdinaryCallingFormat(),
+    )
+
+    if action == 'create':
+            create_bucket(conn, bucket_name)
+
+    if action == 'list':
+            list_bucket(conn)
+
+    if action == 'delete':
+            delete_bucket(conn, bucket_name)
+
+    if action == 'upload_data':
+        upload_data(conn, bucket_name)
+
+    if action == 'download_data':
+        download_data(conn, user, bucket_name)
+{%endace%}
+
+The script needs the following environment variables:
+
+* `HOST` - Hostname of the machine running the RGW service in the cluster.
+* `PORT` - Port of the RGW service.
+* `<USER>_ACCESS_KEY` - USER's `ACCESS_KEY`
+* `<USER>_SECRET_KEY` - USER's `SECRET_KEY`
+
+We previously created a service to provide external access to the RGW.
+
+```bash
+kubectl -n rook-ceph get service rook-ceph-rgw-my-store-external
+NAME                              TYPE       CLUSTER-IP       EXTERNAL-IP   PORT(S)        AGE
+rook-ceph-rgw-my-store-external   NodePort   10.111.198.148   <none>        80:31765/TCP   2h
+```
+Internally the RGW service is running on port `80`. The external port in this case is `31765`.
+
+Set the `HOST` and `PORT` environment variables:
+
+```bash
+export HOST=$(minikube ip)
+export PORT=$(kubectl -n rook-ceph get service rook-ceph-rgw-my-store-external -o jsonpath='{.spec.ports[?(@.name=="rgw")].nodePort}')
+```
+
+Get Alice's and Bob's `ACCESS_KEY` and `SECRET_KEY` from the Kubernetes Secret and set the following environment variables:
+
+```bash
+export ALICE_ACCESS_KEY=$(kubectl get secret rook-ceph-object-user-my-store-alice -n rook-ceph -o yaml | grep AccessKey | awk '{print $2}' | base64 --decode)
+export ALICE_SECRET_KEY=$(kubectl get secret rook-ceph-object-user-my-store-alice -n rook-ceph -o yaml | grep SecretKey | awk '{print $2}' | base64 --decode)
+export BOB_ACCESS_KEY=$(kubectl get secret rook-ceph-object-user-my-store-bob -n rook-ceph -o yaml | grep AccessKey | awk '{print $2}' | base64 --decode)
+export BOB_SECRET_KEY=$(kubectl get secret rook-ceph-object-user-my-store-bob -n rook-ceph -o yaml | grep SecretKey | awk '{print $2}' | base64 --decode)
+```
+
+Now let's create a bucket and add some data to it.
+
+* First, `Bob` creates a bucket `supersecretbucket`
+
+    ```bash
+    python s3test.py Bob create supersecretbucket
+    ```
+
+* List the bucket just created
+
+    ```bash
+    python s3test.py Bob list
+    ```
+
+    The output will be something like:
+
+    ```bash
```

Comment:
`bash` is a bit odd here, too. Since it's plain command output, I think we could just use this:
```suggestion
    ```
```

Reply:
Hrmm could we use ` ```raw `, or something like that? `bash` just seems odd. However, it also doesn't matter much, let's ignore it 😄

Reply:
`raw` makes sense.  Updated the doc.